tsdb/engine/tsm1/file_store.go

```diff
@@ -635,16 +641,20 @@ func (f *FileStore) replace(oldFiles, newFiles []string, updatedFn func(r []TSMF
 	f.mu.RUnlock()
 
 	updated := make([]TSMFile, 0, len(newFiles))
+	tsmTmpExt := fmt.Sprintf("%s.%s", TSMFileExtension, TmpTSMFileExtension)
 
 	// Rename all the new files to make them live on restart
 	for _, file := range newFiles {
 		var newName = file
-		if strings.HasSuffix(file, ".tmp") {
+		if strings.HasSuffix(file, tsmTmpExt) {
 			// The new TSM files have a tmp extension.  First rename them.
 			newName = file[:len(file)-4]
 			if err := os.Rename(file, newName); err != nil {
 				return err
 			}
+		} else if !strings.HasSuffix(file, TSMFileExtension) {
```

Comment:
I think this should be `TmpTSMFileExtension`.  New files passed in are always tmp one.

Reply:
That check happens in the `if` above. It checks for `tsmTmpExt` which is `.tsm.tmp`. This `else if` skips any non-tsm or non-tmp-tsm files (such as index files).

In the `if` above we rename the `tmp` tsm file and drop the `.tmp`. Then below this `else if` we process the `tsm` file.

Without this `else if` to skip non-tsm files I believe we could get caught up on `tsi` index files.